compiler/rustc_typeck/src/check/wfcheck.rs

```diff
@@ -193,13 +193,18 @@ pub fn check_item_well_formed(tcx: TyCtxt<'_>, def_id: LocalDefId) {
 pub fn check_trait_item(tcx: TyCtxt<'_>, def_id: LocalDefId) {
     let hir_id = tcx.hir().local_def_id_to_hir_id(def_id);
     let trait_item = tcx.hir().expect_trait_item(hir_id);
+    let mut span = trait_item.span;
 
     let method_sig = match trait_item.kind {
         hir::TraitItemKind::Fn(ref sig, _) => Some(sig),
+        hir::TraitItemKind::Type(_bounds, Some(ty)) => {
+            span = ty.span;
+            None
+        }
         _ => None,
     };
```

Comment:
Nit but this could be
```rust
let (method_sig, span) = match trait_item.kind {
    hir::TraitItemKind::Fn(ref sig, _) => (Some(sig), trait_item.span),
    hir::TraitItemKind::Type(_bounds, Some(ty)) => (None, ty.span),
    _ => (None, trait_item.span),
};
```
and same below